annotto-front/src/modules/project/components/common/image/ZoomMark.js

```diff
@@ -0,0 +1,210 @@
+import { flatten, isNumber, min, max } from 'lodash'
+import PropTypes from 'prop-types'
+import { useMemo, useRef, useState, useEffect } from 'react'
+import { Group, Image, Line, Text, Transformer } from 'react-konva'
+import useImage from 'use-image'
+
+import theme from '__theme__'
+
+const ZoomMark = ({
+  index,
+  tasks,
+  annotationIndex,
+  predictionIndex,
+  zone,
+  value,
+  imageWidth,
+  imageHeight,
+  isSelected,
+  onValidateClick,
+  onDeleteClick,
+  onSelectClick,
+  onTransformEnd,
+  ...props
+}) => {
+  const lineRef = useRef()
+  const trRef = useRef()
+
+  const [isHovered, setIsHovered] = useState(false)
+
+  const [tagIcon] = useImage(`${process.env.PUBLIC_URL}/static/images/tag.svg`)
+  const [patternIcon] = useImage(`${process.env.PUBLIC_URL}/static/images/pattern.svg`)
+  const [closeIcon] = useImage(`${process.env.PUBLIC_URL}/static/images/close.svg`)
+  const [validateIcon] = useImage(`${process.env.PUBLIC_URL}/static/images/validate.svg`)
+
+  const isPrefill = isNumber(annotationIndex) && isNumber(predictionIndex)
+  const isPrediction = isNumber(predictionIndex) && !isNumber(annotationIndex)
+  const isEditable = !isPrediction && !isPrefill
+
+  const points = useMemo(
+    () => flatten(zone.map(({ x, y }) => [x * imageWidth, y * imageHeight])),
+    [zone, imageHeight, imageWidth]
+  )
+
+  const task = useMemo(() => {
+    const currentTask = tasks.find((t) => t.value === value)
+    if (currentTask && !currentTask?.color) {
+      currentTask.color = theme.colors.defaultAnnotationColors[tasks.indexOf(currentTask)]
+    }
+
+    return currentTask
+  }, [tasks, value])
+
+  const tagPosition = useMemo(
+    () => ({
+      minX: min(zone.map(({ x }) => x * imageWidth)),
+      minY: min(zone.map(({ y }) => y * imageHeight)),
+      maxX: max(zone.map(({ x }) => x * imageWidth)),
+    }),
+    [zone, imageWidth, imageHeight]
+  )
+
+  const styleZoom = useMemo(() => {
+    switch (true) {
+      case isHovered:
+        return { fill: task?.color, opacity: 0.3 }
+      case isPrediction:
+        return { fillPatternImage: patternIcon, opacity: 0.5, dash: [4, 4] }
+      case isPrefill:
+        return { fill: 'transparent', opacity: 0.5, dash: [4, 4] }
+
+      default:
+        return { fill: 'transparent', opacity: 1 }
+    }
+  }, [isHovered, task, isPrediction, isPrefill])
+
+  useEffect(() => {
+    if (isSelected && trRef.current && lineRef.current) {
+      trRef.current.nodes([lineRef.current])
+      trRef.current.getLayer().batchDraw()
+    }
+  }, [isSelected, trRef.current, lineRef.current])
+
+  const _onMouseEnter = () => setIsHovered(true)
+
+  const _onMouseLeave = () => setIsHovered(false)
+
+  if (!imageHeight && !imageWidth) {
+    return null
+  }
+
+  return (
+    <Group draggable={isEditable} onMouseEnter={_onMouseEnter} onMouseLeave={_onMouseLeave} {...props}>
```

Comment:
I think we should avoid as much as possible spreading props since it make the code less readable (and maybe one day we'll switch the codebase in TS, and it won't be allowed)